scripts/partition_manager/partition_manager.rst

```diff
@@ -0,0 +1,408 @@
+      before: list
+         Place the partition before the first existing partition in the list.
+
+      after: list
+         Place the partition after the first existing partition in the list.
+
+   Valid values in the lists are ``app``, ``start``, ``end``, or the name of any partition.
+   It is not possible to place the partition after ``end`` or before ``start``.
+
+
+span: list
+   This property lists which partitions this partition should span across.
+   Partitions with this property are phony partitions.
+   Therefore, this property cannot be used together with the ``placement`` property.
+
+   The Partition Manager fails if a span is empty or if the partitions are not consecutive after processing.
+   Non-existing partitions are ignored.
+
+   .. note::
+      You can specify configurations with an ambiguous ordering (see the following examples).
+      However, different versions of the script might produce a different ordering for such configurations, and the Partition Manager might fail to find a solution even if one is theoretically possible.
+      The Partition Manager always detects unsatisfiable configurations (no false positives), but it might fail on some valid inputs (false negatives).
+
+   See the following examples of valid and invalid configurations:
+
+   .. code-block:: yaml
+      :caption: Span example 1 (fixed order, cannot work)
+
+      mcuboot:
+         placement:
+            before: [spm, app]
+
+      spm:
+         placement:
+            before: [app]
+
+      foo:
+         span: [mcuboot, app] # This will fail, because 'spm' will be placed between mcuboot and app.
+
+      # Order: mcuboot, spm, app
+
+   .. code-block:: yaml
+      :caption: Span example 2 (ambiguous order)
+
+      mcuboot:
+         placement:
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app] # The order of spm and app is ambiguous in this case, but since
+                              # this span exists, Partition Manager will try to increase the
+                              # likelihood that mcuboot and app are placed next to each other.
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # The algorithm should coerce order 2 to make foo work.
+
+   .. code-block:: yaml
+      :caption: Span example 3 (ambiguous order, cannot work)
+
+      mcuboot:
+         placement:
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app]
+
+      bar:
+         span: [mcuboot, spm]
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # foo requires order 2, while bar requires order 1.
+
+inside: list
+   This property is the inverse of ``span``.
+   ``partition_name`` is added to the ``span`` list of the first existing phony partition in the list.
```

Comment:
Is "first existing *phony* partition" correct? And that it only applies to image and placeholder partitions?

Reply:
Yes that is correct.